### PR TITLE
hedgedhttp: bump cristalhq/hedgedhttp

### PR DESCRIPTION
**Pull request by cristaloleg**
**What this PR does**:
Updates cristalhq/hedgedhttp library to use new API. Old is still present but probably will be removed in the future.

**Which issue(s) this PR fixes**:

Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by aknuds1 — COMMENTED** [review]
Is it documented anywhere that `hedgedhttp.NewRoundTripper` is deprecated ("old API")? I can't see it in the [docs](https://pkg.go.dev/github.com/cristalhq/hedgedhttp).

**Comment by cristaloleg** [comment]
Hey, 

> Is it documented anywhere that hedgedhttp.NewRoundTripper

Currently not but as the `cristalhq/hedgedhttp` author I can confirm that deprecation will happen in the next release. Not because it's wrong or bad, just because `hedgedhttp.New` (and `hedgedhttp.Client` as a result) implements `http.RoundTripper`.

> Also, can this be moved out of draft mode?

I haven't updated CHANGELOG file and haven't verified are there other things to do to get accepted. If it looks good for you - you/me can change the status.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Changelog
 
+* [CHANGE] hedgedhttp: bump cristalhq/hedgedhttp #418
```
Comment (aknuds1) [inline CHANGELOG.md:3]: Would you mind just dropping this change? Shouldn't be necessary, as there shouldn't be any functional difference.

```suggestion
```

**Review by aknuds1 — COMMENTED** [review]
AFAICT, the bump is straightforward, but I don't think a changelog entry is warranted as it shouldn't make any functional difference.

**Review by aknuds1 — APPROVED** [review]
Thanks!

**Pull request merged**